From: Niko Wilbert <mail@nikowilbert.de>
Subject: FIX: fixed broken inclusion of CSS data

Index: mdp-toolkit/bimdp/hinet/bihtmlvisitor.py
===================================================================
--- mdp-toolkit.orig/bimdp/hinet/bihtmlvisitor.py
+++ mdp-toolkit/bimdp/hinet/bihtmlvisitor.py
@@ -108,7 +108,7 @@ def show_biflow(flow, filename=None, tit
         html_file = open(filename, 'w')
     html_file.write('<html>\n<head>\n<title>%s</title>\n' % title)
     html_file.write('<style type="text/css" media="screen">')
-    html_file.write(mdp.utils.BASIC_STYLE)
+    html_file.write(mdp.utils.basic_css())
     html_file.write(BiHiNetHTMLVisitor.hinet_css())
     html_file.write('</style>\n</head>\n<body>\n')
     html_file.write('<h3>%s</h3>\n' % title)
